cirq-core/cirq/experiments/parallel_readout_calibration.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Iterable, List, Optional, TYPE_CHECKING
+
+import time
+import random
+
+import numpy as np
+import sympy
+
+from cirq import circuits, ops, study
+from cirq.experiments.readout_experiment_result import ReadoutExperimentResult
+
+if TYPE_CHECKING:
+    import cirq
+
+
+def estimate_parallel_readout_errors(
+    sampler: 'cirq.Sampler',
+    *,
+    qubits: Iterable['cirq.Qid'],
+    trials: int = 20,
+    repetitions: int = 1000,
+    trials_per_batch: Optional[int] = None,
+    bit_strings: Optional[List[int]] = None,
+) -> ReadoutExperimentResult:
+
+    """Estimate readout error for qubits simultaneously.
+
+    For each trial, prepare a bitstring of random |0〉and |1〉states for
+    each state.  Measure each qubit.  Capture the errors per qubit of
+    zero and one state over each triel.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubits: The qubits being tested.
+        repetitions: The number of measurement repetitions to perform for
+            each trial.
+        trials: The number of bitstrings to prepare.
+        trials_per_batch:  If provided, split the experiment into batches
+            with this number of trials in each batch.
+        bit_strings: A list of ints that specifies the bit strings for each
```

Comment:
very strange that these are raveled into a 1d list, and backwards to what I would have expected. Why not keep as a 2d array?

Reply:
Changed to 2d numpy array.